src/Streamly/Data/Fold.hs

```diff
@@ -28,15 +27,14 @@
 -- >>> Data.List.foldl' (+) 0 [1..100]
 -- 5050
 --
--- 'Fold's have an early termination capability e.g. the 'head' fold would
--- terminate on an infinite stream:
---
--- >>> Stream.fold Fold.head $ Stream.fromList [1..]
--- Just 1
+-- 'Fold's are non-greedy consumers and have the have the capability to
+-- terminate early. Being non-greedy, they only consume the part of the input
+-- stream that is required.
 --
--- The above example is similar to the following right fold:
```

Comment:
It is not. It is similar to a terminating `foldl`. With respect to the behavior it is similar to `foldr` I guess but I don't know if it's the correct analogy to use here.